src/toxcore/tcp/codec.rs

```diff
@@ -361,29 +362,14 @@ mod tests {
         // not enough bytes to decode Packet
         assert!(bob_codec.decode(&mut buf).unwrap().is_none());
     }
-    #[test]
-    fn decode_packet_error() {
```

Comment:
Why did you remove this test?

Reply:
Because it's broken. It meant to test deserialization with connection id = 0 but it doesn't. It fails on decryption step due to wrong codec with wrong key being used. Also, connection id = 0 is valid case now.

Reply:
But you can test it with id=1

Reply:
ok, rewrote this test completely with id = 0x0F (1 is also valid packet id).